local/network.go

```diff
@@ -448,11 +445,15 @@ func (ln *localNetwork) Healthy(ctx context.Context) chan error {
 		return healthyChan
 	}
 
-	nodes := make([]*localNode, 0, len(ln.nodes))
-	for _, node := range ln.nodes {
-		nodes = append(nodes, node)
-	}
 	go func() {
+		ln.lock.RLock()
+		defer ln.lock.RUnlock()
+
+		nodes := make([]*localNode, 0, len(ln.nodes))
```

Comment:
in this situation there should be no need to copy ln.nodes to nodes.

Reply:
fixed